internal/persistence/json/database.go

```diff
@@ -129,3 +149,49 @@ func (db *Database) write() error {
 	}
 	return nil
 }
+
+func (db *Database) MigrateToV1(providers []provider.Provider) {
```

Comment:
Let's have this called when creating the database within `NewDatabase` instead of from `main()`

We can then also unexport it as `migrateToV1`